### PR TITLE
Support unicode escapes in TextFormat.

### PR DESCRIPTION
**Pull request by thomasvl**
NOTE: The existing JSON support handles surrogate pairs differently that the
conformance test for TextFormat. This implementation follow the conformance test
recommended handing even though it differs from what the C++ implementation
does.

Also update known failing conformance tests list.

### CODE REVIEW

**Comment by thomasvl** [comment]
Just look at the first commit to see the real changes needed for this.

File: Sources/SwiftProtobuf/TextFormatScanner.swift [hunk 1]
```diff
@@ -116,6 +124,58 @@ private func decodeString(_ s: String) -> String? {
...
+          // \u - 4 hex digits, \U 8 hex digits:
+          if let digit1 = bytes.next(),
+              let d1 = uint32FromHexDigit(digit1),
+              let digit2 = bytes.next(),
+              let d2 = uint32FromHexDigit(digit2),
+              let digit3 = bytes.next(),
+              let d3 = uint32FromHexDigit(digit3),
+              let digit4 = bytes.next(),
+              let d4 = uint32FromHexDigit(digit4) {
+            var codePoint = (d1 << 12) + (d2 << 8) + (d3 << 4) + d4
```
Comment (tbkka) [inline Sources/SwiftProtobuf/TextFormatScanner.swift:137]: Personally, I might factor out the process of grabbing the next four bytes by using a local function that accepted the iterator.  But I might not.  😉   Up to you.

File: Tests/SwiftProtobufTests/Test_Conformance.swift [hunk 2]
```diff
@@ -77,13 +77,57 @@ class Test_Conformance: XCTestCase, PBTestHelpers {
...
+        }
+
+        assertTextFormatDecodeFails("optional_string: \"\\u")
+        assertTextFormatDecodeFails("optional_string: \"\\uDC\"")
+        assertTextFormatDecodeFails("optional_string: \"\\uDCXY\"")
+        assertTextFormatDecodeFails("optional_string: \"\\U")
+        assertTextFormatDecodeFails("optional_string: \"\\UDC\"")
+        assertTextFormatDecodeFails("optional_string: \"\\UDCXY\"")
+        assertTextFormatDecodeFails("optional_string: \"\\U1234DC\"")
+        assertTextFormatDecodeFails("optional_string: \"\\U1234DCXY\"")
```
Comment (tbkka) [inline Tests/SwiftProtobufTests/Test_Conformance.swift:95]: Be nice to verify that too-short escapes also fail to decode at the end of a string:

  " \u123"
  "\U1234567"

etc.

**Review by tbkka — APPROVED** [review]
Looks good!

File: Tests/SwiftProtobufTests/Test_Conformance.swift (hunk 2 above)
Comment (thomasvl) [reply Tests/SwiftProtobufTests/Test_Conformance.swift:95]: Isn't that what `\UDC` and \`U1234DC` check?

Comment (tbkka) [reply Tests/SwiftProtobufTests/Test_Conformance.swift:95]: Oh.  Yeah.  Good point.

File: Sources/SwiftProtobuf/TextFormatScanner.swift (hunk 1 above)
Comment (thomasvl) [reply Sources/SwiftProtobuf/TextFormatScanner.swift]: Done.  Inserted a new commit to do it so the diff is pretty easy to look at.

**Pull request merged**